### ISSUE TITLE
Sporadic segfault when solving the same MIP

### ISSUE THREAD

**Issue by grinya007** [ERGO-Code/HiGHS#1345]
Hi!

I'm experiencing sporadic memory-related failures in a relatively simple MIP.
An example MPS file https://gist.github.com/grinya007/7219af99a70e9d5b7d3b8a3b677d4e9b
I built highs 1.5.3 and when I run it against this file it would fail with `corrupted size vs. prev_size` or `malloc(): invalid next size (unsorted)` in ~10% of runs. 90% of runs solve the MIP successfully. I'm not sure what more details are required but would be glad to answer any questions.

Thank you!

**Comment by jajhall** [comment]
Thanks. I've identified an illegal write when solving your problem that may well be the cause of this. 

I'll try to track down the cause of this

**Comment by grinya007** [comment]
Hi @jajhall! does PR #1355 fix this issue or is it a typo in the commit message?

**Comment by jajhall** [comment]
No, #1355 refers to #1354. The reference to #1345 in an earlier commit was a typo.

Sorry to have got you excited for nothing!

**Comment by fwesselm** [comment]
Hi!

I have debugged this a little bit and, as far as I understand, a free column is eliminated using a cut / row that is later deleted (see HighsPostsolveStack::FreeColSubstitution::undo in HightPostsolveStack.cpp).

Therefore, when the postsolve is performed, the 'row' index is not valid anymore. I have limited knowledge of the code base, so there may be a more elegant way to fix this, but I was able to make it work by just checking the row index.

I suspect that similar issues may occur in the other postsolve reductions.

[FreeColSubstitution.txt](https://github.com/ERGO-Code/HiGHS/files/12311698/FreeColSubstitution.txt)

**Comment by jajhall** [comment]
Great @fwesselm , this gives me more to go on. I'm busy this week and next - after a very busy summer - and then life calms down, so I'll get back to MIP issues

**Comment by grinya007** [comment]
Dear @galabovaa, the commit that you are referring to is unrelated to this issue, it was a mistake in the commit message. Please, let's reopen this so that some time it will be fixed as the problem is already acknowledged.

**Comment by galabovaa** [comment]
Sure!

**Comment by jajhall** [comment]
This is fixed by #1525

**Comment by khantute** [comment]
Hi! I'm facing the same issue described here but with the new release 1.13 with the attached model. It seems like I only have the "corrupted size vs. prev_size" message and it's apparearing for every run.

[pricing_model.zip](https://github.com/user-attachments/files/25073114/pricing_model.zip)

**Comment by jajhall** [comment]
Now that the LP file can be read (see #2812), I can look at the issue!

**Comment by jajhall** [comment]
This is fixed by https://github.com/ERGO-Code/HiGHS/pull/1525
